crates/primitives/src/proofs.rs

```diff
@@ -70,18 +70,55 @@ pub fn calculate_withdrawals_root(withdrawals: &[Withdrawal]) -> B256 {
 
 /// Calculates the receipt root for a header.
 pub fn calculate_receipt_root(receipts: &[ReceiptWithBloom]) -> B256 {
+    #[cfg(feature = "optimism")]
+    {
+        // There is a minor bug in op-geth and op-erigon where in the Regolith hardfork,
+        // the receipt root calculation does not include the deposit nonce in the receipt
+        // encoding. This will be fixd in the next hardfork, however for now, we must strip
+        // the deposit nonce from the receipts before calculating the receipt root.
```

Comment:
Is there an issue to track this in op-geth? If not we should make one ASAP and ensure it's priod

Reply:
There is. This bug is fixed in Canyon, #5210